doc/examples/transform/plot_tps_deformation.py

```diff
@@ -0,0 +1,119 @@
+sometimes we need more flexibility. One model that represents a non-linear
+transformation, i.e. one where lines can bend, is thin-plate splines [1]_ [2]_.
+
+Thin-plate splines draw on the analogy of a metal sheet, which has inherent
+rigidity. Consider our source points: each has to move a certain distance, in
+both the x and y directions, to land in their corresponding target positions.
+First, examine only the x coordinates. Imagine placing a thin metal plate on
+top of the image. Now bend it, such that at each source point, the plate's
+z offset is the distance, positive or negative, that that source point has to
+travel in the x direction in order to land in its target position. The plate
+resists bending, and therefore remains smooth. We can read offsets for
+coordinates other than source points from the position of the plate. The same
+procedure can be repeated for the y coordinates.
+
+This gives us our thin-plate spline model that maps any (x, y) coordinate to a
+target position.
+
+.. [1] Wikipedia, Thin plate spline
+       https://en.wikipedia.org/wiki/Thin_plate_spline
+
+.. [2] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+       decomposition of deformations." IEEE Transactions on pattern analysis and
+       machine intelligence 11.6 (1989): 567–585.
+       :DOI:`10.1109/34.24792`
+       https://user.engineering.uiowa.edu/~aip/papers/bookstein-89.pdf
+
+
+Deform an image
+===============
+
+In this example, we demonstrate how to correct a barrel distortion [3]_ with a
+transform based on thin-plate splines. A barrel distortion create a "fisheye"
+like effect, where the image magnification decreases with the distance from the
+image center (optical axis).
+
+Let's demonstrate the non-linear effect and its correction with the image of
+a checkerboard, which we first warp accordingly.
+
+.. [3] https://en.wikipedia.org/wiki/Distortion_(optics)#Radial_distortion
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import skimage as ski
+
+
+def radial_distortion(xy, k1=0.9, k2=0.5):
+    """Distort coordinates `xz` symmetrically around their own center."""
+    xy_c = xy.max(axis=0) / 2
+    xy = (xy - xy_c) / xy_c
+    radius = np.linalg.norm(xy, axis=1)
+    distortion_model = (1 + k1 * radius + k2 * radius**2) * k2
+    xy *= distortion_model.reshape(-1, 1)
+    xy = xy * xy_c + xy_c
+    return xy
+
+
+image = ski.data.checkerboard()
+image = ski.transform.warp(image, radial_distortion, cval=0.5)
+
+
+# Pick points in `src` by hand and move their corresponding points in `dst` closer
+# to their expected position.
+# fmt: off
+src = np.array([[22,  22], [100,  10], [177, 22], [190, 100], [177, 177], [100, 188],
+                [22, 177], [ 10, 100], [ 66, 66], [133,  66], [ 66, 133], [133, 133]])
+dst = np.array([[ 0,   0], [100,   0], [200,  0], [200, 100], [200, 200], [100, 200],
+                [ 0, 200], [  0, 100], [ 73, 73], [128,  73], [ 73, 128], [128, 128]])
```

Comment:
You could pick by hand, or you could apply the inverse warp directly. But not sure how to find the formula for that, so I think this is probably better :)